sorts/cyclic_sort.py

```diff
@@ -0,0 +1,52 @@
+"""
+This is a pure Python implementation of the Cyclic Sort algorithm.
+
+For doctests run following command:
+python -m doctest -v cyclic_sort.py
+or
+python3 -m doctest -v cyclic_sort.py
+For manual testing run:
+python cyclic_sort.py
+"""
+
+
+def cyclic_sort(nums: list) -> list:
+    """
+    Sorts the input list in-place using the Cyclic Sort algorithm.
+
+    :param nums: List of integers to be sorted.
+    :return: The same list sorted in ascending order.
+
+    Time complexity: O(n), where n is the number of elements in the list.
+
+    Examples:
+    >>> cyclic_sort([3, 5, 2, 1, 4])
+    [1, 2, 3, 4, 5]
+    >>> cyclic_sort([])
+    []
+    """
+
```

Comment:
This would be a good place to ensure the inputs meet the constraints of the algorithm. If they don't, indicate to the user what the issue is with the provided list.